### ISSUE TITLE
_result inconsistency

### ISSUE THREAD

**Issue by hennevogel** [openSUSE/open-build-service#6924]
```
/build/OBS:Server:Unstable/_result?&package=obs-server&arch=x86_64&repository=SLE_12_SP4
```

A query for an nonexistent project, package or repository throws 404's. A query for  an nonexistent architecture returns an empty result list.

**Comment by adrianschroeter** [comment]
yes, because the project is the object and the arch is only the filter parameter.

**Comment by hennevogel** [comment]
the package is also just the filter parameter...

**Comment by mlschroe** [comment]
The difference is that the arch list is a property of the repository, so it is allowed to specify an architecture that is not present in some of the repositories.

I could throw an error if a architecture is not present in any of the repository, though. Do you think that's worth to implement?

**Comment by hennevogel** [comment]
Well it avoids [stuff like this](https://github.com/openSUSE/obs-tools/blob/master/obs-package-status/status-check.rb#L34) in API consumers...
